nemo_curator/modules/splitter.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pandas as pd
+
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.modules.base import BaseModule
+
+
+class DocumentSplitter(BaseModule):
+    """
+    Splits documents into segments based on a separator.
+    Each segment is a new document with an additional column
+    indicating the segment id.
+
+    To restore the original document, ensure that each document
+    has a unique id prior to splitting.
+    """
+
+    def __init__(
+        self,
+        separator: str,
+        text_field: str = "text",
+        segment_id_field: str = "segment_id",
+    ):
+        """
+        Args:
+            separator (str): The separator to split the documents on.
+            text_field (str): The name of the column containing the text to split.
+            segment_id_field (str): The name of the column to add to indicate the segment id.
+        """
+        super().__init__(input_backend="any")
+        self.separator = separator
+        self.text_field = text_field
+        self.segment_id_field = segment_id_field
+
+    def _split_partition(self, df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Given that the backend is any, this should be :

```suggestion
    def _split_partition(self, df: Union[pd.DataFrame, cudf.DataFrame]) -> Union[pd.DataFrame, cudf.DataFrame]:
```

Also import cuDF using gpu_only_import above